docs/devguide/xtypes.rst

```diff
@@ -1042,6 +1042,13 @@ The operations used for this include the DynamicData operations named "set_*" fo
 They work similarly to the "get_*" operations that are described in section :ref:`xtypes--interpreting-data-samples-with-dynamicdata`.
 When populating the DynamicData of complex data types, use get_complex_value() (see :ref:`xtypes--reading-members-of-more-complex-types`) to navigate from DynamicData representing containing types to DynamicData representing contained types.
 
+Unions have the following semantics in the OpenDDS implementation of DynamicData.
+Unions start in an "empty" state meaning that no branch is selected.
+At the point of serialization, the middleware will treat an empty union as one that has the default branch selected and provide a corresponding default value.
+If the union does not have a default branch defined, then the results are undefined.
```

Comment:
Could this be "implementation-defined" instead of "undefined." Maybe it needs more explanation than that - is it going to be some valid discriminator?

Reply:
Currently, it works as follows during serialization. If the union is empty, the discriminator will get the default value of its type. If that value doesn't select any branch, then only the discriminator is serialized. If it selects a branch, the default value for that branch is also serialized.

Now that I reviewed the serialization code for union, it can be simplified a bit to remove some unnecessary logic. I'll go ahead to update that.